### ISSUE TITLE
Support block parameter in UnusedMethodParameter

### ISSUE THREAD

**Issue by gsampaio** [oclint/oclint#135]
I have just found a unused parameter false positive. It seems that arguments passed inside a block are considered not considered on the Unused Parameter rule. 

I have this [gist](https://gist.github.com/gsampaio/9ef36ed9aac0b28af5c1) to illustrate the problem. 

On the function run_on_main_queue() i just dispatch a block with a parameter to the main queue. When I run the oclint i get this output

```
OCLint Report

Summary: TotalFiles=1 FilesWithViolations=1 P1=0 P2=0 P3=1 

/Users/gsampaio/Desktop/UnusedArgument/UnusedArgument/main.m:3:37: unused method parameter P3 

[OCLint (http://oclint.org) v0.7]
```

Thanks

**Comment by ryuichis** [comment]
This is not a false positive after all. Because the way to define that block should be `void(^block)(NSInteger)` instead of `void(^block)(NSInteger arg)`. Based on the sample code you provided, `arg` will never be used. And in case you by accident use that variable, like this

```objective-c
#include <stdio.h>

void run_on_main_queue(void(^block)(int arg), int argument)
{
    printf("%d", arg);
    block(argument);
}
```

When you compile the code, the compiler will throw errors like
```bash
block.m:5:18: error: use of undeclared identifier 'arg'
    printf("%d", arg);
                 ^
```

So hope we can conclude that `arg` is an unused method parameter, and will be impossible to be used.

**Comment by gsampaio** [comment]
Makes sense, thanks :)
